docs/source/reference/types.rst

```diff
@@ -0,0 +1,286 @@
+
+.. code:: idris
+
+    vectorMap : {n : Nat} -> (a -> b) -> Vect n a -> Vect n b
+
+We would then use ``vectorMap`` as follows:
+
+.. code-block:: idris
+
+    vectorMap f v
+
+or
+
+.. code:: idris
+
+    vectorMap {n = m} f v
+
+These two examples show the power of Idris implicits: while they can be inferred by the compiler, if necessary (or desired) they may be specified by the user
+
+Idris has *two* additional modifiers of implicits, however.
+The first, while not that important to the type system of Idris, is *incredibly* useful, and it is the default implicit.
+Essentially "regular" implicits only try to figure out the type without adding any extra information.
+
+Default implicits, on the other hand, instead of trying to figure out a value, will instead just use a default, which means that unnecessary "simple" functions are unnecessary in Idris: you just use the default value.
+
+The syntax for default implicits is ``{default val x : a} -> b``, where ``val : a``, which is the default value for ``x``.
+
+The third kind of implicits are auto implicits.
+Auto implicits are most useful in proofs, where they differ from regular implicits by there purpose.
+Regular implicits try to find a "necessary" match.
+``auto`` implicits, on the other hand, look for *any* possible construction of a value.
+This is useful for propositions, where we often don't need to reason about the specific value of a type member, merely that it exists.
+Essentially, auto implicits are for cases where we want the compiler to have more freedom to infer the value
+
+For instance, if we want to automatically pass a proof through some functions, we use ``auto`` defaults.
+
+.. hint::
+    Lowercase names in types are automatically universally quantified in types.
+    For instance, ``map : (a -> b) -> List a -> List b`` is transformed into ``map : {0 a : Type} -> {0 b : Type} -> (a -> b) -> List a -> List b`` [#f1]_.
+    You can write this signature explicitly, and it will compile the same as the simple version, and in addition you could also have the ``forall`` keyword and wrote this as ``map : forall a . forall b . ((a -> b) -> List a -> List b)``.
+
+    If desired, this behavior can be toggled off by ``%unbound_implicits off`` (and back on by ``%unbound_implicits on``)
+
+
+
+^^^^^^^^^^^^^
+Quantities
+^^^^^^^^^^^^^
+
+Idris 2 uses Quantitative Type Theory (`QTT`_) as its theoretical basis.
+In QTT, every function type has an associated multiplicity, ``0``, ``1``, or ω.
+These correspond with how many times the given input is "used" in the result type.
+
+Very roughly, ``0`` corresponds to a value "being ignored" or *erased*, a value of ``1`` corresponds to a value being "used" once, or being *one-to-one*, and ω (which, in Idris syntax is omitted) corresponds to a type having an unrestricted number of uses.
+
+Note that this idea of "usage" is not always cut and dry.
```

Comment:
Usage is cut and dry. If your variable appears at runtime then it's used, erased quantities cannot appear at runtime, period.

The only loose rules are around subtyping of ω arguments and linearity but that shouldn't be something the audience of this text needs to worry about.